sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java

```diff
@@ -68,7 +83,24 @@ public EventGridEvent(String subject, String eventType, Object data, String data
      * @throws NullPointerException if eventGridJsonString is {@code null}.
      */
     public static List<EventGridEvent> fromString(String eventGridJsonString) {
-        return EventGridDeserializer.deserializeEventGridEvents(eventGridJsonString);
+        try {
+            return Arrays.stream(DESERIALIZER
+                .deserialize(new ByteArrayInputStream(eventGridJsonString.getBytes(StandardCharsets.UTF_8)),
+                    TypeReference.createInstance(com.azure.messaging.eventgrid.implementation.models.EventGridEvent[].class)))
+                .map(internalEvent -> {
+                    if (internalEvent.getSubject() == null || internalEvent.getEventType() == null
+                        || internalEvent.getData() == null) {
+                        throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
shouldn't they be NullPointerException as you are checking for null ?

Reply:
Same as above, this is to check attributes of a json string whether it has null attributes. I think IllegalArgumentException might be better because the json string itself isn't null.